packages/react-composites/tests/browser/chat/app/FakeAdapterApp.tsx

```diff
@@ -180,3 +193,51 @@ const sendRemoteFileSharingMessage = (chatClientModel: Model, remoteParticipant:
     }
   );
 };
+
+const initializeAdapters = async (
+  participants: ChatParticipant[],
+  chatClientModel: Model,
+  thread
+): Promise<ChatAdapter[]> => {
+  const remoteAdapters = [];
+  for (const participant of participants) {
+    const remoteChatClient = new FakeChatClient(chatClientModel, participant.id);
+    const remoteAdapter = await initializeAdapter({
+      userId: participant.id,
+      displayName: participant.displayName,
+      chatClient: remoteChatClient as IChatClient as ChatClient,
+      chatThreadClient: remoteChatClient.getChatThreadClient(thread.chatThread?.id ?? 'INVALID_THREAD_ID')
+    });
+    remoteAdapters.push(remoteAdapter);
+  }
+  return remoteAdapters;
+};
+
+const createHiddenComposites = (remoteAdapters: ChatAdapter[]): JSX.Element[] => {
+  return remoteAdapters.map((remoteAdapter) => {
+    const userId = toFlatCommunicationIdentifier(remoteAdapter.getState().userId);
+    return (
+      <div key={`composite-${userId}`} style={{ height: 0, overflow: 'hidden' }}>
+        <_IdentifierProvider identifiers={generateIDS(userId)}>
+          <ChatComposite
+            adapter={remoteAdapter}
+            options={{
+              participantPane: true
+            }}
+          />
+        </_IdentifierProvider>
+      </div>
+    );
+  });
+};
+
+/**
+ * Helper function to create IDs for {@link _IdentifierProvider} using a user id
```

Comment:
This is a little worrysome, its not obvious that if I add a data-ui-id somewhere in the DOM, it'll suddenly be appended by the participants userId - meaning I might write a UI test and have it fail not because the UI test couldn't find my element, but because the data-ui-id I specified was changed to be appended with the userID.
This also only applies when we use the IdentifierProvider which we've moved away from and are just specifying data-ui-ids typically as hardcoded strings directly in the DOM.

Consider having the hidden composites having the id of `hidden-composite-{userId}` instead, such that the query selector can then do something like:
`#hidden-composite-{userId} data-ui-id=[typing-indicator]` meaning its a totally transparent change for anyone not making use of the hidden-composites.

Either that or change _all_ data-ui-id selectors used throughout all tests to be appended with the userId

Reply:
Great idea! Removed `IdentifierProvider` from the hidden composites and using the id as a qualifier for the selector. Thanks.